shared/middleware/helpers/animation_bucket.rb

```diff
@@ -15,4 +15,37 @@ def allowed_file_types
   def cache_duration_seconds
     3600
   end
+
+  #
+  # Extends the default 'get' operation in BucketHelper.  If an animation is
+  # requested by version, and the animation is not found, try again without
+  # a version (just getting the latest animation).
+  # If that works, log the incident since we likely had a bad version id
+  # and we want to see these incidents go down over time.
+  #
+  def s3_get_object(key, if_modified_since, version)
+    super(key, if_modified_since, version)
+  rescue Aws::S3::Errors::NoSuchVersion
+    # Rethrow if it clearly wasn't the version.
+    raise if version.nil?
+
+    # Locate the latest non-deleted version
+    versions = s3.list_object_versions(bucket: @bucket, prefix: key).versions
+    raise if versions.empty?
+
+    # Try getting the first (non-delete-marker) version
+    s3_object = super(key, if_modified_since, versions.first.version_id)
+
+    # If the fallback is successful, let's notify Honeybadger, because we'd
+    # like these to go down over time.
+    Honeybadger.notify(
+      'AnimationBucket served latest version instead of requested version',
+      context: {
+        s3_key: key,
+        requested_version: version,
+        served_version: s3_object.version_id
+      }
+    )
+    s3_object
+  end
```

Comment:
I decided against a special-case Honeybadger notification when the latest version is deleted, because from an implementation point of view both paths are identical - and what we really care about is roughly measuring the frequency of bad version id requests, and seeing that volume go down over time.

Reply:
Agreed, assuming I'm understanding what you're saying here -- the thing that is definitely bad is if the requested version is missing. whether the latest version then turns out to be deleted or not seems less important.